packages/scxa-experiment-page/src/TSnePlotViewRoute.js

```diff
@@ -0,0 +1,338 @@
+            let isMetadataCluster = parseInt(colourByValue) ? false : true
+            if(isMetadataCluster) {
+              this.setState({
+                selectedColourBy : colourByValue,
+                selectedColourByCategory : METADATA_PLOT
+              })
+              updateUrlWithParams( [{colourBy: colourByValue}])
+            } else {
+              this.setState({
+                selectedClusterId : colourByValue,
+                selectedColourByCategory : CLUSTERS_PLOT
+              })
+              updateUrlWithParams([{k: colourByValue}])
+            }
+           }
+          }
+          onChangeMarkerGeneFor={(selectedOption) => {
+            this.setState((state) => ({
+              selectedClusterIdOption: selectedOption
+            }))
+          }}
+          ksWithMarkers={ksWithMarkerGenes}
+          metadata={metadata.map(data => {return {value: data.value.replaceAll(`_`, ` `), label: data.label}})}
+          species={species}
+          heatmapType={this.state.selectedColourByCategory === METADATA_PLOT ? CELL_TYPE_MARKER_GENE_HEATMAP : CLUSTER_MARKER_GENE_HEATMAP}
+        />
+      },
+      {
+        path: `/anatomogram`,
+        title: `Anatomogram`,
+        main: () =>
+          <AnatomogramCellTypeHeatmapView
+            showIds={anatomogram[organWithMostOntologies]}
+            experimentAccession={experimentAccession}
+            species={species}
+            organ={organWithMostOntologies}
+            host={atlasUrl}
+          />
+      },
+      {
+        path: `/gene-info`,
+        title: `Gene information`,
+        main: () =>
+          <BioentityInformationWithFetchLoader
+            host={atlasUrl}
+            resource={`json/bioentity-information/${search.geneId}`}/>
+      }
+    ]
+
+    const updateUrl = (query) => {
+      history.push({...history.location, search: query.toString()})
+    }
+
+    const resetHighlightClusters = (query) => {
+      if(query.has(`clusterId`)) {
+        query.delete(`clusterId`)
+      }
+    }
+
+    const updateUrlWithParams = (params) => {
+      const query = new URLSearchParams(history.location.search)
+      params.forEach(param => query.set(Object.keys(param)[0], Object.values(param)[0]))
+      resetHighlightClusters(query)
+      updateUrl(query)
+    }
+
+    const getPlotOption = (selectedPlotType) => {
+      return Object.keys(plotTypeDropdown[plotTypeDropdown.findIndex(
+          (plot) => plot.plotType.toLowerCase() ===  selectedPlotType)].plotOptions[0])[0] + `: ` + search.plotOption
+    }
+
+    const preferredK = this.props.selectedK ?
+        this.props.selectedK.toString() :
+          this.props.ks.length > 0 ?
+          this.props.ks[0].toString() : ``
+
+    const basename = URI(`experiments/${experimentAccession}${match.path}`, URI(atlasUrl).path()).toString()
+
+    const sideTabStyle = {overflow: `clip`, textOverflow: `ellipsis`}
+    return (
```

Comment:
I would insert an empty line above this `return` statement to make the code more readable.
Thanks

Reply:
Please wrap the code into a small function to improve readability.